### PR TITLE
[SPARK-6198][SQL] Support "select current_database()"

### PR DESCRIPTION
**Pull request by pzzs**
The method(evaluate) has changed in UDFCurrentDB, it just throws a exception.But hiveUdfs call this method and failed.
@Override
public Object evaluate(DeferredObject[] arguments) throws HiveException
{ throw new IllegalStateException("never");

### CODE REVIEW

**Comment by DoingDone9** [comment]
anyone will test it ? @marmbrus @srowen

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/sqlUDFCurrentDB.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.apache.hadoop.hive.ql.session.SessionState
+
+// deterministic in the query range
+@Description(name = "current_database",
+    value = "_FUNC_() - returns currently using database name")
+class sqlUDFCurrentDB extends UDFCurrentDB {
+
+  // This function just throws an exception in hive0.13
+  override def evaluate(arguments: Array[DeferredObject]): Object = {
+    SessionState.get().getCurrentDatabase()
```
Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/sqlUDFCurrentDB.scala]: Does this actually work in the distributed mode?

**Comment by DoingDone9** [comment]
yes, it works.  I have tested it in the distributed mode with two nodes. @rxin

**Comment by marmbrus** [comment]
@DoingDone9, really?  In my tests it null pointers when you try to get the session state on an executor.  It seems like they made this change in Hive 13 on purpose because you should not be accessing session state from an executor.

If this is something we really want to support, I think we should instead add a rule to `prepareForExecution` that rewrite this expression statically with the current database.  Until then I propose we close this issue.

**Comment by DoingDone9** [comment]
my previous test was successful and i will test it again. Thank you @marmbrus

**Comment by DoingDone9** [comment]
Could you tell me how you got this exception? I test with three nodes , and it works again. Thank you @marmbrus

**Comment by marmbrus** [comment]
Here is the command I ran:

```
sc.parallelize(1 to 10).map(_ => org.apache.hadoop.hive.ql.session.SessionState.get().getCurrentDatabase()).collect()
```

Independent of whether this works in some configurations I still think computing it statically before execution is the right thing to do.

**Comment by DoingDone9** [comment]
I do not  agree that. Because this expression  is foldable, then it will be computed  in ConstantFolding of Optimizer. So I will get the name of currentDB after optimizer not after  execution. like that 
```
== Analyzed Logical Plan ==
Project [HiveGenericUdf#org.apache.spark.sql.hive.sqlUDFCurrentDB() AS _c0#59]
 NoRelation$

== Optimized Logical Plan ==
Project [default AS _c0#59]
 NoRelation$
```

@marmbrus

**Comment by marmbrus** [comment]
Ah, I see.  Can you add comments that explain that and reopen this PR?

**Comment by DoingDone9** [comment]
I have opened a new pr https://github.com/apache/spark/pull/5538

**Pull request closed without merging**